common-rest/src/main/java/org/uniprot/api/rest/output/converter/JsonMessageConverter.java

```diff
@@ -78,7 +85,32 @@ protected void before(MessageConverterContext<T> context, OutputStream outputStr
     @Override
     protected void writeEntity(T entity, OutputStream outputStream) throws IOException {
         JsonGenerator generator = tlJsonGenerator.get();
-        generator.writeObject(projectEntryFields(entity));
+        List<ReturnField> fields = getThreadLocalFilterMap();
+        if (Utils.notNullNotEmpty(fields)) {
+            ObjectMapper mapper = objectMapper.copy();
```

Comment:
do we need to do a copy? jackson recommends to reuse objectmappers because they're threadsafe (also see this quick google on stackoverflow https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field)

Reply:
I am manipulating the object mapper and it will have different filters, depending on each request. I am afraid it will keep the previous request behaviour, this is why I copied. I am implementing a test for it. will updated this comment when get the result